File: backend/src/schemas/lead.py
"""
Lead Pydantic schemas for request/response validation.

Defines DTOs for lead submission and retrieval.
Validates all inputs at API boundaries before processing.
"""

import re
from datetime import datetime, date
from typing import Optional, List
from uuid import UUID

from pydantic import BaseModel, Field, EmailStr, field_validator, model_validator

from ..models.lead import (
    ConditionType,
    DurationType,
    TreatmentType,
    UrgencyType,
    PriorityType,
    LeadStatus,
    ContactMethodType,
    ContactOutcome,
)


# =============================================================================
# UTM Parameters Schema
# =============================================================================

class UTMParams(BaseModel):
    """
    UTM tracking parameters from widget embedding.
    
    Captures marketing attribution data from URL parameters.
    """
    
    utm_source: Optional[str] = Field(
        default=None,
        max_length=255,
        description="Traffic source (e.g., google, facebook)"
    )
    utm_medium: Optional[str] = Field(
        default=None,
        max_length=255,
        description="Marketing medium (e.g., cpc, email)"
    )
    utm_campaign: Optional[str] = Field(
        default=None,
        max_length=255,
        description="Campaign name"
    )
    utm_term: Optional[str] = Field(
        default=None,
        max_length=255,
        description="Paid search keywords"
    )
    utm_content: Optional[str] = Field(
        default=None,
        max_length=255,
        description="Content identifier for A/B testing"
    )

    model_config = {
        "json_schema_extra": {
            "example": {
                "utm_source": "google",
                "utm_medium": "cpc",
                "utm_campaign": "tms_awareness_2024",
                "utm_term": "depression treatment",
                "utm_content": "ad_variant_a"
            }
        }
    }


# =============================================================================
# Lead Creation Schema (Widget Submission)
# =============================================================================

class LeadCreate(BaseModel):
    """
    Schema for creating a new lead from widget submission.
    
    Validates all intake form fields before processing.
    PHI fields will be encrypted before storage.
    """
    
    # Contact Information (PHI)
    first_name: str = Field(
        ...,
        min_length=1,
        max_length=100,
        description="Patient first name (required)"
    )
    last_name: Optional[str] = Field(
        default=None,
        max_length=100,
        description="Patient last name (optional)"
    )
    email: EmailStr = Field(
        ...,
        description="Patient email address (required)"
    )
    phone: str = Field(
        ...,
        min_length=10,
        max_length=20,
        description="Patient phone number (required)"
    )
    
    # Date of Birth (for age verification - TMS requires 18+)
    date_of_birth: Optional[date] = Field(
        default=None,
        description="Patient date of birth (optional, used for age verification)"
    )
    
    # Clinical Information
    condition: ConditionType = Field(
        ...,
        description="Primary mental health condition"
    )
    condition_other: Optional[str] = Field(
        default=None,
        max_length=500,
        description="Description if condition is 'OTHER'"
    )
    
    # Multi-condition support (NEW)
    conditions: Optional[List[ConditionType]] = Field(
        default=None,
        description="List of conditions (multi-select)"
    )
    other_condition_text: Optional[str] = Field(
        default=None,
        max_length=500,
        description="Description for 'OTHER' condition"
    )
    
    # Severity Assessments (NEW - conditional based on conditions)
    # PHQ-2 for Depression (0-3 scale each)
    phq2_interest: Optional[int] = Field(
        default=None,
        ge=0, le=3,
        description="PHQ-2: Little interest or pleasure (0-3)"
    )
    phq2_mood: Optional[int] = Field(
        default=None,
        ge=0, le=3,
        description="PHQ-2: Feeling down, depressed, hopeless (0-3)"
    )
    
    # GAD-2 for Anxiety (0-3 scale each)
    gad2_nervous: Optional[int] = Field(
        default=None,
        ge=0, le=3,
        description="GAD-2: Feeling nervous, anxious (0-3)"
    )
    gad2_worry: Optional[int] = Field(
        default=None,
        ge=0, le=3,
        description="GAD-2: Not being able to stop worrying (0-3)"
    )
    
    # OCD severity (1-4 scale)
    ocd_time_occupied: Optional[int] = Field(
        default=None,
        ge=1, le=4,
        description="OCD: Time occupied by obsessions (1-4)"
    )
    
    # PTSD severity (0-4 scale)
    ptsd_intrusion: Optional[int] = Field(
        default=None,
        ge=0, le=4,
        description="PTSD: Intrusion symptoms (0-4)"
    )
    
    # TMS therapy interest (NEW)
    tms_therapy_interest: Optional[str] = Field(
        default=None,
        max_length=50,
        description="TMS therapy interest (daily_tms, accelerated_tms, saint_protocol, not_sure)"
    )
    
    # Preferred contact method (NEW)
    preferred_contact_method: Optional[str] = Field(
        default=None,
        max_length=20,
        description="Preferred contact method (phone_call, text, email, any)"
    )
    
    symptom_duration: DurationType = Field(
        ...,
        description="How long symptoms have been present"
    )
    prior_treatments: List[TreatmentType] = Field(
        default=[],
        description="List of prior treatments tried"
    )
    
    # Insurance Information
    has_insurance: bool = Field(
        ...,
        description="Whether patient has health insurance"
    )
    insurance_provider: Optional[str] = Field(
        default=None,
        max_length=255,
        description="Insurance provider name if has_insurance=true"
    )
    other_insurance_provider: Optional[str] = Field(
        default=None,
        max_length=255,
        description="Other insurance provider name"
    )
    
    # Location
    zip_code: str = Field(
        ...,
        min_length=5,
        max_length=10,
        description="Patient ZIP code for service area check"
    )
    
    # Urgency & Consent
    urgency: UrgencyType = Field(
        ...,
        description="How urgently patient wants treatment"
    )
    hipaa_consent: bool = Field(
        ...,
        description="Patient acknowledged HIPAA consent"
    )
    sms_consent: bool = Field(
        default=False,
        description="Patient consents to SMS communication"
    )
    
    # UTM Tracking (optional)
    utm_params: Optional[UTMParams] = Field(
        default=None,
        description="Marketing attribution parameters"
    )
    
    # Referral Information (NEW - matches Jotform)
    is_referral: Optional[bool] = Field(
        default=None,
        description="Whether patient was referred by a healthcare provider"
    )
    referring_provider_name: Optional[str] = Field(
        default=None,
        max_length=255,
        description="Name of referring healthcare provider"
    )
    referring_provider_specialty: Optional[str] = Field(
        default=None,
        max_length=255,
        description="Specialty of referring healthcare provider (e.g., Psychiatrist, Psychologist)"
    )
    referring_clinic: Optional[str] = Field(
        default=None,
        max_length=255,
        description="Clinic/practice name of referring provider"
    )
    referring_provider_email: Optional[str] = Field(
        default=None,
        max_length=255,
        description="Email of referring healthcare provider"
    )
    
    # Metadata (set by backend)
    referrer_url: Optional[str] = Field(
        default=None,
        max_length=2000,
        description="Page URL where widget was embedded"
    )
    
    # ==========================================================================
    # Validators
    # ==========================================================================
    
    @field_validator("phone")
    @classmethod
    def validate_phone(cls, v: str) -> str:
        """
        Validate and normalize phone number.
        
        Accepts international phone numbers from any country.
        Supports formats like:
        - +1 (555) 123-4567 (US)
        - +44 20 7946 0958 (UK)
        - +254 712 345 678 (Kenya)
        - +91 98765 43210 (India)
        - +86 138 0013 8000 (China)
        
        Normalizes to E.164 format (digits with optional leading +).
        """
        # Remove common formatting characters but preserve leading +
        has_plus = v.strip().startswith('+')
        cleaned = re.sub(r"[\s\-\(\)\.]", "", v)
        
        # Extract just digits
        digits_only = re.sub(r"[^\d]", "", cleaned)
        
        # Validate minimum length (shortest valid numbers are ~7 digits)
        if len(digits_only) < 7:
            raise ValueError("Phone number must be at least 7 digits")
        
        # Maximum E.164 is 15 digits
        if len(digits_only) > 15:
            raise ValueError("Phone number too long (max 15 digits)")
        
        # Return normalized format with + if it had one
        if has_plus:
            return f"+{digits_only}"
        return digits_only
    
    @field_validator("zip_code")
    @classmethod
    def validate_zip_code(cls, v: str) -> str:
        """
        Validate US ZIP code format.
        
        Accepts 5-digit or 9-digit (ZIP+4) formats.
        """
        # Remove spaces and dashes
        cleaned = v.replace(" ", "").replace("-", "")
        
        if not cleaned.isdigit():
            raise ValueError("ZIP code must contain only digits")
        
        if len(cleaned) not in (5, 9):
            raise ValueError("ZIP code must be 5 or 9 digits")
        
        return cleaned[:5]  # Store only first 5 digits
    
    @field_validator("first_name", "last_name")
    @classmethod
    def validate_name(cls, v: Optional[str]) -> Optional[str]:
        """
        Validate and sanitize name fields.
        
        Removes extra whitespace and validates characters.
        """
        if v is None:
            return v
        
        # Strip and normalize whitespace
        v = " ".join(v.split())
        
        # Basic validation - allow letters, spaces, hyphens, apostrophes
        if not re.match(r"^[a-zA-Z\s\-\'\.]+$", v):
            raise ValueError("Name contains invalid characters")
        
        return v
    
    @model_validator(mode="after")
    def validate_conditional_fields(self) -> "LeadCreate":
        """
        Validate fields that depend on other field values.
        """
        # condition_other OR other_condition_text required if condition is OTHER
        # Accept either field for backward compatibility
        has_other_text = self.condition_other or self.other_condition_text
        
        # Check if OTHER is in conditions array or is the primary condition
        has_other_condition = (
            self.condition == ConditionType.OTHER or 
            (self.conditions and ConditionType.OTHER in self.conditions)
        )
        
        if has_other_condition and not has_other_text:
            raise ValueError("condition_other or other_condition_text is required when condition is 'OTHER'")
        
        # HIPAA consent must be given
        if not self.hipaa_consent:
            raise ValueError("HIPAA consent is required to submit")
        
        return self

    model_config = {
        "json_schema_extra": {
            "example": {
                "first_name": "John",
                "last_name": "Doe",
                "email": "john.doe@example.com",
                "phone": "(555) 123-4567",
                "condition": "DEPRESSION",
                "symptom_duration": "MORE_THAN_12_MONTHS",
                "prior_treatments": ["ANTIDEPRESSANTS", "THERAPY_CBT"],
                "has_insurance": True,
                "insurance_provider": "Blue Cross Blue Shield",
                "zip_code": "85001",
                "urgency": "ASAP",
                "hipaa_consent": True,
                "sms_consent": True,
                "utm_params": {
                    "utm_source": "google",
                    "utm_medium": "cpc"
                }
            }
        }
    }


# =============================================================================
# Lead Update Schema
# =============================================================================

class LeadUpdate(BaseModel):
    """
    Schema for updating an existing lead.
    
    All fields are optional - only provided fields will be updated.
    PHI fields will be re-encrypted if modified.
    """
    
    # Contact Information (PHI) - optional updates
    first_name: Optional[str] = Field(
        default=None,
        min_length=1,
        max_length=100,
        description="Patient first name"
    )
    last_name: Optional[str] = Field(
        default=None,
        max_length=100,
        description="Patient last name"
    )
    email: Optional[EmailStr] = Field(
        default=None,
        description="Patient email address"
    )
    phone: Optional[str] = Field(
        default=None,
        max_length=20,
        description="Patient phone number"
    )
    
    # Clinical Information
    condition: Optional[ConditionType] = Field(
        default=None,
        description="Primary mental health condition"
    )
    condition_other: Optional[str] = Field(
        default=None,
        max_length=500,
        description="Description if condition is 'OTHER'"
    )
    symptom_duration: Optional[DurationType] = Field(
        default=None,
        description="How long symptoms have been present"
    )
    prior_treatments: Optional[List[TreatmentType]] = Field(
        default=None,
        description="List of prior treatments tried"
    )
    
    # Insurance Information
    has_insurance: Optional[bool] = Field(
        default=None,
        description="Whether patient has health insurance"
    )
    insurance_provider: Optional[str] = Field(
        default=None,
        max_length=255,
        description="Insurance provider name"
    )
    
    # Location
    zip_code: Optional[str] = Field(
        default=None,
        max_length=10,
        description="Patient ZIP code"
    )
    
    # Urgency
    urgency: Optional[UrgencyType] = Field(
        default=None,
        description="How urgently patient needs treatment"
    )
    
    # Notes
    notes: Optional[str] = Field(
        default=None,
        max_length=5000,
        description="Coordinator notes about the lead"
    )
    
    # Status and Priority
    status: Optional[LeadStatus] = Field(
        default=None,
        description="Current lead status"
    )
    priority: Optional[PriorityType] = Field(
        default=None,
        description="Lead priority level"
    )

    @field_validator('phone')
    @classmethod
    def normalize_phone(cls, v: Optional[str]) -> Optional[str]:
        """Normalize phone to E.164 format if provided."""
        if v is None:
            return None
        # Remove common formatting characters but preserve leading +
        if v.startswith('+'):
            cleaned = '+' + ''.join(c for c in v[1:] if c.isdigit())
        else:
            cleaned = ''.join(c for c in v if c.isdigit())
            # Add US country code if not present
            if len(cleaned) == 10:
                cleaned = '+1' + cleaned
            elif not cleaned.startswith('1') and len(cleaned) == 11:
                cleaned = '+' + cleaned
            else:
                cleaned = '+' + cleaned
        return cleaned

    model_config = {
        "json_schema_extra": {
            "example": {
                "first_name": "Jane",
                "email": "jane.doe@example.com",
                "notes": "Patient requested morning appointments only",
                "priority": "HOT"
            }
        }
    }


# =============================================================================
# Lead Response Schemas
# =============================================================================

class LeadResponse(BaseModel):
    """
    Schema for lead data returned to authenticated dashboard users.
    
    Includes decrypted PHI for authorized access only.
    """
    
    id: UUID
    
    # Contact Info (decrypted for authorized users)
    first_name: str
    last_name: Optional[str] = None
    email: str
    phone: str
    
    # Clinical Info
    condition: ConditionType
    condition_other: Optional[str] = None
    symptom_duration: DurationType
    prior_treatments: List[TreatmentType]
    
    # Insurance
    has_insurance: bool
    insurance_provider: Optional[str] = None
    
    # Location
    zip_code: str
    in_service_area: bool
    
    # Urgency & Consent
    urgency: UrgencyType
    hipaa_consent: bool
    hipaa_consent_timestamp: Optional[datetime] = None
    privacy_consent_timestamp: Optional[datetime] = None
    sms_consent: bool
    sms_consent_timestamp: Optional[datetime] = None
    
    # Scoring
    score: int
    priority: PriorityType
    
    # Status
    status: LeadStatus
    notes: Optional[str] = None
    
    # UTM
    utm_source: Optional[str] = None
    utm_medium: Optional[str] = None
    utm_campaign: Optional[str] = None
    
    # Timestamps
    created_at: datetime
    updated_at: datetime
    contacted_at: Optional[datetime] = None
    
    # Scheduling fields
    scheduled_callback_at: Optional[datetime] = None
    scheduled_notes: Optional[str] = None
    contact_method: Optional[ContactMethodType] = None
    last_contact_attempt: Optional[datetime] = None
    contact_attempts: Optional[int] = 0
    next_follow_up_at: Optional[datetime] = None
    
    # Contact Outcome
    contact_outcome: ContactOutcome = ContactOutcome.NEW
    
    # Follow-up tracking
    follow_up_reason: Optional[str] = Field(
        default=None,
        description="Reason for follow-up (e.g., 'No Answer', 'Not Interested', 'No Show')"
    )
    follow_up_date: Optional[datetime] = Field(
        default=None,
        description="When to follow up with this lead"
    )
    
    # Last Activity timestamp
    last_updated_at: Optional[datetime] = Field(
        default=None,
        description="Timestamp of last modification to this lead (NULL for new untouched leads)"
    )

    model_config = {
        "from_attributes": True
    }


class LeadListResponse(BaseModel):
    """
    Schema for lead list items (minimal data for table display).
    
    Used in dashboard lead table for performance.
    Includes phone and email for coordinator outreach.
    """
    
    id: UUID
    lead_number: Optional[str] = None
    first_name: str
    last_name: Optional[str] = None
    # Contact info for coordinator outreach
    email: str
    phone: str
    condition: ConditionType
    # Multi-condition support
    conditions: Optional[List[str]] = None
    other_condition_text: Optional[str] = None
    # Preferred contact method
    preferred_contact_method: Optional[str] = None
    score: int
    priority: PriorityType
    status: LeadStatus
    in_service_area: bool
    created_at: datetime
    scheduled_callback_at: Optional[datetime] = None
    
    # Contact Outcome for coordinator tracking
    contact_outcome: ContactOutcome = ContactOutcome.NEW
    contact_attempts: Optional[int] = 0
    last_contact_attempt: Optional[datetime] = None
    
    # Referral tracking fields
    is_referral: bool = False
    referring_provider_id: Optional[UUID] = None
    referring_provider_name: Optional[str] = None
    
    # Follow-up tracking
    follow_up_reason: Optional[str] = Field(
        default=None,
        description="Reason for follow-up (e.g., 'No Answer', 'Not Interested', 'No Show')"
    )
    
    # Last Activity timestamp
    last_updated_at: Optional[datetime] = Field(
        default=None,
        description="Timestamp of last modification to this lead (NULL for new untouched leads)"
    )

    model_config = {
        "from_attributes": True
    }


# =============================================================================
# Contact Outcome Schemas
# =============================================================================

class UpdateContactOutcomeRequest(BaseModel):
    """
    Schema for updating a lead's contact outcome.
    
    Used by coordinators to track outreach results.
    """
    
    contact_outcome: ContactOutcome = Field(
        ...,
        description="Result of contact attempt"
    )
    notes: Optional[str] = Field(
        default=None,
        max_length=1000,
        description="Notes about the contact outcome"
    )
    next_follow_up_at: Optional[datetime] = Field(
        default=None,
        description="Next scheduled follow-up (for NO_ANSWER, CALLBACK_REQUESTED)"
    )

    model_config = {
        "json_schema_extra": {
            "example": {
                "contact_outcome": "NO_ANSWER",
                "notes": "Called twice, no answer. Will try again tomorrow.",
                "next_follow_up_at": "2026-01-23T10:00:00Z"
            }
        }
    }


# =============================================================================
# Scheduling Schemas
# =============================================================================

class ScheduleCallbackRequest(BaseModel):
    """
    Schema for scheduling a coordinator callback OR consultation.
    
    CRITICAL ROUTING LOGIC:
    - schedule_type='callback': Sets contact_outcome=CALLBACK_REQUESTED, keeps in Follow-up Queue
    - schedule_type='consultation': Sets status=SCHEDULED, contact_outcome=SCHEDULED, moves to Scheduled Queue
    """
    
    scheduled_callback_at: datetime = Field(
        ...,
        description="Date and time for the scheduled callback/consultation"
    )
    scheduled_notes: Optional[str] = Field(
        default=None,
        max_length=1000,
        description="Notes for the scheduled callback/consultation"
    )
    contact_method: ContactMethodType = Field(
        default=ContactMethodType.PHONE,
        description="Preferred method of contact"
    )
    schedule_type: Optional[str] = Field(
        default="callback",
        description="Type of schedule: 'callback' (stays in follow-up) or 'consultation' (moves to scheduled)"
    )

    model_config = {
        "json_schema_extra": {
            "example": {
                "scheduled_callback_at": "2026-01-22T10:00:00Z",
                "scheduled_notes": "Patient prefers morning calls",
                "contact_method": "PHONE",
                "schedule_type": "callback"
            }
        }
    }


class LogContactAttemptRequest(BaseModel):
    """
    Schema for logging a contact attempt.
    """
    
    contact_method: ContactMethodType = Field(
        ...,
        description="Method used to contact the lead"
    )
    was_successful: bool = Field(
        ...,
        description="Whether the contact attempt was successful"
    )
    notes: Optional[str] = Field(
        default=None,
        max_length=1000,
        description="Notes about the contact attempt"
    )
    next_follow_up_at: Optional[datetime] = Field(
        default=None,
        description="Next scheduled follow-up if not reached"
    )

    model_config = {
        "json_schema_extra": {
            "example": {
                "contact_method": "PHONE",
                "was_successful": False,
                "notes": "No answer, left voicemail",
                "next_follow_up_at": "2026-01-23T14:00:00Z"
            }
        }
    }


class ScheduledLeadResponse(BaseModel):
    """
    Schema for scheduled leads in calendar view.
    """
    
    id: UUID
    lead_number: Optional[str] = None
    first_name: str
    last_name: Optional[str] = None
    condition: ConditionType
    priority: PriorityType
    status: LeadStatus
    scheduled_callback_at: datetime
    scheduled_notes: Optional[str] = None
    contact_method: Optional[ContactMethodType] = None
    contact_attempts: Optional[int] = 0
    phone: Optional[str] = None  # Decrypted for authorized users

    model_config = {
        "from_attributes": True
    }


class LeadSubmitResponse(BaseModel):
    """
    Schema for widget submission response.
    
    Returns minimal data to confirm submission without exposing internals.
    """
    
    success: bool = Field(
        ...,
        description="Whether submission was successful"
    )
    message: str = Field(
        ...,
        description="User-friendly confirmation message"
    )
    lead_id: Optional[UUID] = Field(
        default=None,
        description="ID of created lead (for tracking)"
    )
    priority: Optional[PriorityType] = Field(
        default=None,
        description="Calculated priority level"
    )
    estimated_response_time: Optional[str] = Field(
        default=None,
        description="Estimated time for clinic to respond"
    )

    model_config = {
        "json_schema_extra": {
            "example": {
                "success": True,
                "message": "Thank you! A care coordinator will contact you within 24 hours.",
                "lead_id": "123e4567-e89b-12d3-a456-426614174000",
                "priority": "HOT",
                "estimated_response_time": "Within 24 hours"
            }
        }
    }

File: frontend/src/types/lead.ts
/**
 * Lead type definitions for NeuroReach AI
 * 
 * Shared across all lead-related components.
 */

/**
 * Lead priority levels for TMS patient qualification
 */
export type LeadPriority = 'hot' | 'medium' | 'low' | 'disqualified';

/**
 * Lead status in the pipeline
 * Must match backend LeadStatus enum exactly
 */
export type LeadStatus = 
  | 'new' 
  | 'contacted' 
  | 'scheduled' 
  | 'consultation complete'  // Backend uses CONSULTATION_COMPLETE
  | 'treatment started'       // Backend uses TREATMENT_STARTED
  | 'lost' 
  | 'disqualified';

/**
 * Contact outcome for coordinator outreach tracking
 * Tracks the result of each contact attempt
 */
export type ContactOutcome = 
  | 'NEW'                 // Not contacted yet
  | 'ANSWERED'            // Spoke with lead, can proceed to schedule
  | 'NO_ANSWER'           // Called but no pickup, needs follow-up
  | 'UNREACHABLE'         // Wrong number, disconnected, etc.
  | 'CALLBACK_REQUESTED'  // Lead asked to call back at specific time
  | 'SCHEDULED'           // Consultation has been scheduled
  | 'COMPLETED'           // Consultation completed successfully
  | 'NOT_INTERESTED';     // Lead declined, archive

/**
 * Contact outcome configuration for UI display
 * Uses Lucide icon names for production-ready SVG icons
 */
export const CONTACT_OUTCOME_CONFIG: Record<ContactOutcome, {
  label: string;
  iconName: 'Sparkles' | 'CheckCircle2' | 'PhoneMissed' | 'PhoneOff' | 'Clock' | 'Ban' | 'Calendar' | 'CheckCircle';
  color: string;
  bgColor: string;
  borderColor: string;
  description: string;
  iconColor: string;
  ringColor: string;
}> = {
  NEW: {
    label: 'New',
    iconName: 'Sparkles',
    color: 'text-blue-700',
    bgColor: 'bg-blue-50',
    borderColor: 'border-blue-200',
    description: 'Not contacted yet',
    iconColor: 'text-blue-500',
    ringColor: 'ring-blue-400',
  },
  ANSWERED: {
    label: 'Answered',
    iconName: 'CheckCircle2',
    color: 'text-emerald-700',
    bgColor: 'bg-emerald-50',
    borderColor: 'border-emerald-200',
    description: 'Spoke with lead, ready to schedule',
    iconColor: 'text-emerald-500',
    ringColor: 'ring-emerald-400',
  },
  NO_ANSWER: {
    label: 'No Answer',
    iconName: 'PhoneMissed',
    color: 'text-amber-700',
    bgColor: 'bg-amber-50',
    borderColor: 'border-amber-200',
    description: 'Called but no pickup',
    iconColor: 'text-amber-500',
    ringColor: 'ring-amber-400',
  },
  UNREACHABLE: {
    label: 'Unreachable',
    iconName: 'PhoneOff',
    color: 'text-red-700',
    bgColor: 'bg-red-50',
    borderColor: 'border-red-200',
    description: 'Wrong number or disconnected',
    iconColor: 'text-red-500',
    ringColor: 'ring-red-400',
  },
  CALLBACK_REQUESTED: {
    label: 'Callback',
    iconName: 'Clock',
    color: 'text-violet-700',
    bgColor: 'bg-violet-50',
    borderColor: 'border-violet-200',
    description: 'Lead requested callback',
    iconColor: 'text-violet-500',
    ringColor: 'ring-violet-400',
  },
  NOT_INTERESTED: {
    label: 'Not Interested',
    iconName: 'Ban',
    color: 'text-slate-700',
    bgColor: 'bg-slate-100',
    borderColor: 'border-slate-300',
    description: 'Lead declined',
    iconColor: 'text-slate-500',
    ringColor: 'ring-slate-400',
  },
  SCHEDULED: {
    label: 'Scheduled',
    iconName: 'Calendar',
    color: 'text-teal-700',
    bgColor: 'bg-teal-50',
    borderColor: 'border-teal-200',
    description: 'Consultation scheduled',
    iconColor: 'text-teal-500',
    ringColor: 'ring-teal-400',
  },
  COMPLETED: {
    label: 'Completed',
    iconName: 'CheckCircle',
    color: 'text-green-700',
    bgColor: 'bg-green-50',
    borderColor: 'border-green-200',
    description: 'Consultation completed',
    iconColor: 'text-green-500',
    ringColor: 'ring-green-400',
  },
};

/**
 * Primary condition the patient is seeking treatment for
 * Must match backend ConditionType enum exactly
 */
export type ConditionType = 
  | 'DEPRESSION' 
  | 'ANXIETY' 
  | 'OCD' 
  | 'PTSD' 
  | 'OTHER';

/**
 * Complete lead record from database
 */
export interface Lead {
  id: string;
  leadId: string; // Human-readable ID like NR-2026-001
  
  // Personal Info
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  dateOfBirth?: string;
  
  // Clinical
  primaryCondition: ConditionType;
  otherConditionText?: string;
  symptomDuration: string;
  priorTreatments: string[];
  currentMedications: boolean;
  
  // Insurance
  hasInsurance: boolean;
  insuranceProvider?: string;
  
  // Location
  zipCode: string;
  isInServiceArea: boolean;
  
  // Preferences
  desiredStart: 'asap' | 'within_30_days' | 'exploring';
  preferredContactMethod: 'phone' | 'email' | 'sms';
  
  // Scoring
  leadScore: number;
  priority: LeadPriority;
  status: LeadStatus;
  
  // Edit modal compatibility fields (populated from raw API response)
  condition?: string;
  urgency?: string;
  notes?: string;
  
  // Attribution
  utmSource?: string;
  utmMedium?: string;
  utmCampaign?: string;
  
  // Timestamps
  createdAt: string;
  updatedAt: string;
  lastContactedAt?: string;
}

/**
 * Lead for table display (subset of full Lead)
 */
export interface LeadTableRow {
  id: string;
  leadId: string;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  condition: string;
  // Multi-condition support
  conditions?: string[];
  otherConditionText?: string;
  // Preferred contact method
  preferredContactMethod?: string;
  priority: LeadPriority;
  status: LeadStatus;
  submittedAt: string;
  // Contact outcome tracking
  contactOutcome?: ContactOutcome;
  contactAttempts?: number;
  lastContactAttempt?: string;
  scheduledCallbackAt?: string;
  // Next follow-up / requested callback time (distinct from consultation time)
  nextFollowUpAt?: string;
  // Last activity timestamp - tracks when lead was last modified (NULL for untouched leads)
  lastUpdatedAt?: string;
  // Follow-up reason tag (e.g., "No Answer", "Callback Requested", "No Show")
  followUpReason?: string;
  // Referral tracking
  isReferral?: boolean;
  referringProviderName?: string;
  referringProviderId?: string;
}

/**
 * Filters for lead queries
 */
export interface LeadFilters {
  priority?: LeadPriority[];
  status?: LeadStatus[];
  dateFrom?: string;
  dateTo?: string;
  searchQuery?: string;
}

/**
 * Preferred contact method options
 */
export type PreferredContactMethod = 'phone_call' | 'text' | 'email' | 'any';

/**
 * Lead submission data from intake widget
 * Field names must match backend API schema (snake_case)
 * 
 * UPDATED: Now includes all fields for Jotform parity:
 * - Multi-condition support
 * - Severity assessments (PHQ-2, GAD-2, OCD, PTSD)
 * - Preferred contact method
 */
export interface LeadSubmitData {
  first_name: string;
  last_name?: string;
  email: string;
  phone: string;
  date_of_birth?: string;  // Date of Birth in YYYY-MM-DD format
  
  // Primary condition (legacy field for backward compatibility)
  condition: ConditionType;
  condition_other?: string;
  
  // Multi-condition support (NEW - array of conditions)
  conditions?: ConditionType[];
  other_condition_text?: string;
  
  // Severity Assessments (NEW - conditional based on conditions)
  // PHQ-2 for Depression (0-3 scale each)
  phq2_interest?: number | null;
  phq2_mood?: number | null;
  
  // GAD-2 for Anxiety (0-3 scale each)
  gad2_nervous?: number | null;
  gad2_worry?: number | null;
  
  // OCD severity (1-4 scale)
  ocd_time_occupied?: number | null;
  
  // PTSD severity (0-4 scale)
  ptsd_intrusion?: number | null;
  
  // Other intake fields
  symptom_duration: DurationType;
  prior_treatments: TreatmentType[];
  has_insurance: boolean;
  insurance_provider?: string;
  other_insurance_provider?: string;
  zip_code: string;
  urgency: UrgencyType;
  hipaa_consent: boolean;
  sms_consent: boolean;
  
  // Preferred contact method (NEW - REQUIRED)
  preferred_contact_method?: PreferredContactMethod;
  
  // TMS Therapy Interest (NEW - matches Jotform)
  tms_therapy_interest?: TMSInterestType;
  
  // Referral information (NEW - matches Jotform)
  is_referral?: boolean;
  referring_provider_name?: string;
  referring_provider_specialty?: string;
  referring_clinic?: string;
  referring_provider_email?: string;
  
  // UTM tracking
  utm_params?: IUTMParams;
  referrer_url?: string;
}

/**
 * Response from lead submission endpoint
 */
export interface LeadSubmitResponse {
  success: boolean;
  leadId: string;
  priority: LeadPriority;
  message: string;
}

// Backward compatibility aliases for existing components
export type ILeadSubmitResponse = LeadSubmitResponse;
export type PriorityType = LeadPriority;

/**
 * Lead creation data (alias for backward compatibility)
 */
export interface ILeadCreate extends LeadSubmitData {}

/**
 * Lead response from API
 */
export interface ILeadResponse extends Lead {}

/**
 * Lead list item for tables
 */
export interface ILeadListItem extends LeadTableRow {}

/**
 * Paginated response
 */
export interface IPaginatedResponse<T> {
  items: T[];
  total: number;
  page: number;
  /** Backend uses `page_size` */
  page_size: number;
  /** Backend uses `total_pages` */
  total_pages: number;
  /** Backend uses `has_next` */
  has_next: boolean;
  /** Backend uses `has_previous` */
  has_previous: boolean;
}

/**
 * UTM tracking parameters
 * Field names must match backend API schema (snake_case)
 */
export interface IUTMParams {
  utm_source?: string;
  utm_medium?: string;
  utm_campaign?: string;
  utm_term?: string;
  utm_content?: string;
}

/**
 * Duration options for symptom duration
 * Must match backend DurationType enum exactly
 */
export type DurationType = 
  | 'LESS_THAN_6_MONTHS' 
  | 'SIX_TO_TWELVE_MONTHS' 
  | 'MORE_THAN_12_MONTHS';

/**
 * Treatment options for prior treatments
 * Must match backend TreatmentType enum exactly
 */
export type TreatmentType = 
  | 'ANTIDEPRESSANTS' 
  | 'THERAPY_CBT' 
  | 'BOTH' 
  | 'NONE' 
  | 'OTHER';

/**
 * Urgency options for treatment start
 * Must match backend UrgencyType enum exactly
 */
export type UrgencyType = 
  | 'ASAP' 
  | 'WITHIN_30_DAYS' 
  | 'EXPLORING';

/**
 * TMS Therapy Interest options
 * SAINT Protocol only available for Depression
 */
export type TMSInterestType = 
  | 'daily_tms' 
  | 'accelerated_tms' 
  | 'saint_protocol'  // Depression only
  | 'not_sure';

/**
 * Condition labels for display
 */
export const CONDITION_LABELS: Record<ConditionType, string> = {
  DEPRESSION: 'Depression',
  ANXIETY: 'Anxiety',
  OCD: 'OCD',
  PTSD: 'PTSD',
  OTHER: 'Other',
};

/**
 * Duration labels for display
 */
export const DURATION_LABELS: Record<DurationType, string> = {
  LESS_THAN_6_MONTHS: 'Less than 6 months',
  SIX_TO_TWELVE_MONTHS: '6-12 months',
  MORE_THAN_12_MONTHS: 'More than 12 months',
};

/**
 * Treatment labels for display
 */
export const TREATMENT_LABELS: Record<TreatmentType, string> = {
  ANTIDEPRESSANTS: 'Antidepressants',
  THERAPY_CBT: 'Therapy/CBT',
  BOTH: 'Both medication and therapy',
  NONE: 'None',
  OTHER: 'Other treatments',
};

/**
 * Urgency labels for display
 */
export const URGENCY_LABELS: Record<UrgencyType, string> = {
  ASAP: 'As soon as possible',
  WITHIN_30_DAYS: 'Within 30 days',
  EXPLORING: 'Just exploring options',
};
